### DISCUSSION TITLE
Assistance Needed for Integrating Contact Form with EmailJS via CDP

### DISCUSSION THREAD

**Discussion by zaynejaz** [emailjs-com/help#125] [Q&A]
Hello,

I have a contact form on my site that sends the payload data to a CDP (Customer Data Platform). The CDP then transforms this data with a transformation built with JS and sends the output payload to EmailJS via a webhook destination (Header = content-type:application/json). I have a few questions regarding this integration:

- Which REST API URL should I use for this integration? I seem to think it's https://www.emailjs.com/docs/rest-api/send/ but when I send in the payload to test with Postman I get a 400 error with a recurring 504 error (504 error is when I test via the CDP).
- Would anyone know what the payload EmailJS expects to receive based on the URL?

Currently, the payload I am using is as follows: 
{
  "service_id": "service_1234567",
  "template_id": "template_1234567",
  "user_id": "12345_98765431",
  "formData": {
    "accountType": "company",
    "companyName": "CompanyName",
    "email": "test@email.com",
    "first_name": "First",
    "last_name": "Last",
    "userMessage": "This is a sample message",
    "timestamp": "Friday, Jul 5, 2024, 9:10:33 PM",
    "page": "http://localhost:3000",
    "path": "/"
  }
}

Could anyone provide insight into any necessary changes?

Note that when I send this payload to EmailJS from the CDP/Postman, I receive a 400 error along with a recurring 504 error, but I have verified that the service, template, and user IDs are correct.

Any help would be greatly appreciated.

Thanks in advance.

**Comment by xr0master** [comment]
You have to use correct "user_id" value
